src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -95,17 +95,17 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 Workspace.AdviseSolutionEvents(solution);
             }
 
-            LoadComponentsInUIContextOnceSolutionFullyLoaded(cancellationToken);
+            await LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(cancellationToken);
         }
 
-        protected override void LoadComponentsInUIContext(CancellationToken cancellationToken)
+        protected override async Task LoadComponentsInUIContextAsync(CancellationToken cancellationToken)
         {
             ForegroundObject.AssertIsForeground();
```

Comment:
❗️ Don't call `AssertIsForeground` in an asynchronous method. Instance, use `SwitchToMainThreadAsync` to switch to the main thread if needed (the switch is a NOP if the code is already on the main thread).

:memo: This appears a few times in this pull request, but I only commented on it here.